src/components/controls/universal-controls.js

```diff
@@ -0,0 +1,140 @@
+var registerComponent = require('../../core/component').registerComponent;
+var THREE = require('../../lib/three');
+
+var COMPONENT_SUFFIX = '-controls';
+var MAX_DELTA = 0.2; // ms
+var PI_2 = Math.PI / 2;
+
+/**
+ * Universal-controls component.
+ *
+ * Receives input events from device-specific control components, and applies movement and rotation
+ * to the element accordingly.
+ */
+module.exports.Component = registerComponent('universal-controls', {
+
+  dependencies: ['velocity'],
+
+  schema: {
+    enabled: { default: true },
+
+    flyingEnabled: { default: false },
+
+    movementEnabled: { default: true },
+    movementControls: { default: ['gamepad', 'wasd', 'touch'] },
+    movementEasing: { default: 15 }, // m/s2
+    movementAcceleration: { default: 65 }, // m/s2
+
+    rotationEnabled: { default: true },
+    rotationControls: { default: ['hmd', 'gamepad', 'mouse'] },
```

Comment:
On aframe we currently limit rotation to the y-axis on touch and let it unconstrained for mouse. Have you thought on how we should handle individual configuration of controls?